File: apps/import-test-2-admin/src/api/order/OrderWhereInput.ts
import { DateTimeFilter } from "../../util/DateTimeFilter";
import { CustomerWhereUniqueInput } from "../customer/CustomerWhereUniqueInput";
import { FloatNullableFilter } from "../../util/FloatNullableFilter";
import { StringFilter } from "../../util/StringFilter";
import { ProductWhereUniqueInput } from "../product/ProductWhereUniqueInput";
import { IntNullableFilter } from "../../util/IntNullableFilter";

export type OrderWhereInput = {
  createdAt?: DateTimeFilter;
  customer?: CustomerWhereUniqueInput;
  discount?: FloatNullableFilter;
  id?: StringFilter;
  product?: ProductWhereUniqueInput;
  quantity?: IntNullableFilter;
  status?: "PENDING" | "COMPLETED" | "CANCELLED";
  totalPrice?: IntNullableFilter;
  updatedAt?: DateTimeFilter;
};

File: apps/service-with-helm/src/APIs/Order/Dtos/OrderWhereInput.cs
namespace ServiceWithHelm.APIs.Dtos;

public class OrderWhereInput
{
    public DateTime? CreatedAt { get; set; }

    public string? Customer { get; set; }

    public string? Details { get; set; }

    public string? Id { get; set; }

    public double? Price { get; set; }

    public DateTime? UpdatedAt { get; set; }
}
